tools/ci/mkchlog-snippet.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+"""
+python3 mkchlog-snippet.py [-d <outdir>] <repo owner> <repo name> <PR number>
+
+This script generates a changelog snippet for a given pull request in a GitHub
+repository.  The snippet is saved to a file in the `changelog.d` directory by
+default or to another directory specified with the `-d` option.
+
+Requirements:
+- Python 3.7+
+- requests
+"""
+
+from __future__ import annotations
+
+import argparse
+import json
+import os
+import sys
+import textwrap
+from dataclasses import dataclass
+from pathlib import Path
+
+import requests
+
+SEMVER_LABELS = [
+    # (Label name, category name) in descending order of precedence
+    ("semver-major", "Breaking Changes"),
+    ("semver-minor", "Enhancements and New Features"),
+    ("semver-patch", "Bug Fixes"),
+    ("semver-dependencies", "Dependencies"),
+    ("semver-documentation", "Documentation"),
+    ("semver-internal", "Internal"),
+    ("semver-performance", "Performance"),
+    ("semver-tests", "Tests"),
+]
+
+GRAPHQL_API_URL = os.environ.get("GITHUB_GRAPHQL_URL", "https://api.github.com/graphql")
+
+
+@dataclass
+class PullRequest:
+    title: str
+    url: str
+    author: str
+    closed_issues: list[str]
+    labels: set[str]
+
+    @property
+    def category(self) -> str:
+        for lbl, cat in SEMVER_LABELS:
+            if lbl in self.labels:
+                return cat
+        sys.exit("Pull request lacks semver labels")
+
+    def as_snippet(self) -> str:
+        item = self.title
```

Comment:
```suggestion
        item = self.title.strip()
```
to also ensure legit subsequent checks etc.